### PR TITLE
Refactor key interaction tests - Pass 1 - Defork

### PR DESCRIPTION
**Pull request by aicioara**
Part of #2628

Replacing #2755

I will be doing 2 passes because I will get more context (and better sense of best practices) by refactoring other files. However, please flag anything that seems unusual.

As part of this: 
- **Test semantics: hierarchy, naming, testing, but no coverage check**
- Applied best practices from the wiki page
- Factored out common code into `beforeEach()` clauses
- No nested `beforeEach()`es
- Cleaned the console
- Remade hierarchy as shown bellow
screen shot 2015-09-11 at 12 35 35 pm

### CODE REVIEW

**Comment by aicioara** [comment]
Finishing this off from my fork. Since previous PR was all outdated, migrating the comments here.

File: test/interactions/keyInteractionTests.ts [hunk 1]
```diff
@@ -1,302 +1,338 @@
...
+      it("fires callback for key press", () => {
+        assert.strictEqual(keyInteraction.onKeyPress(A_KEY_CODE, aKeyCodeCallback), keyInteraction,
+          "setting the keyPress callback returns the interaction");
+        keyInteraction.attachTo(component);
+
+        TestMethods.triggerFakeMouseEvent("mouseover", component.background(), 100, 100);
+        TestMethods.triggerFakeKeyboardEvent("keydown", component.background(), A_KEY_CODE);
+        assert.isTrue(aKeyCodeCallbackCalled, "callback for \"a\" was called when \"a\" key was pressed");
+
+        keyInteraction.offKeyPress(A_KEY_CODE, aKeyCodeCallback);
```
Comment (aicioara) [inline test/interactions/keyInteractionTests.ts]: | Why this line?

Not entirely sure why removing it kills the future tests. Investigating...

Comment (aicioara) [inline test/interactions/keyInteractionTests.ts]: However, it sounds like a bug and it sounds like outside the scope of this Pull Request. I can make an issue on this.

**Comment by aicioara** [comment]
Ready for review @bluong

File: test/interactions/keyInteractionTests.ts (hunk 1 above)
Comment (bluong) [inline test/interactions/keyInteractionTests.ts]: Please make the issue when you can.

Comment (aicioara) [inline test/interactions/keyInteractionTests.ts]: Opened https://github.com/palantir/plottable/issues/2838

**Pull request merged**